org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/DocumentProviderTest.xtend

```diff
@@ -273,4 +261,248 @@ class DocumentProviderTest extends AbstractGitTest {
 		documentProvider.load(existingImageFile)
 	}
 
+	@Test
+	def void saveWithRemoteChangesCreatesUnversionedBackupFileAndRaisesException() {
+		// given
+		val existingFileName = createPreExistingFileInRemoteRepository
+		val localGit = gitProvider.git
+		localGit.pull.call
+		
+		val localChange = 'Contents of file after local change'
+		
+		val remoteChange = 'Contents of file after remote change'
+		remoteGitFolder.root.write(existingFileName, remoteChange)
+		remoteGit.addAndCommit(existingFileName, "change on remote")
+
+		// when
+		try {
+			documentProvider.save(existingFileName, localChange)
+
+			// then			
+			fail('Expected ConflictingModificationsException, but none was thrown.')
+		} catch (ConflictingModificationsException exception) {
+			val backupFileName = existingFileName + '.local-backup'
```

Comment:
numbering for multiple locally existing backup files?
the filename should actually be retrievable from the result of the save action.
Thus the exception should hold enough information to retrieve the backup file name (if it exists). I'm still unsure whether an exception is appropriate here. We might as well use a return object in the save method that holds all necessary information since we will definitely have to handle these cases. let's talk about that tomorrow.

Reply:
Implemented the enumeration, left the exception in place, as discussed.